tests/scalers/prometheus/azure_managed_prometheus/azure_managed_prometheus_test.go

```diff
@@ -0,0 +1,382 @@
+//go:build e2e
+// +build e2e
+
+package azure_managed_prometheus_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+// Load environment variables from .env file
+var _ = godotenv.Load("../../.env")
+
+const (
+	testNameWorkloadIdentity = "azure-managed-prometheus-workload-identity-test"
+	testNamePodIdentity      = "azure-managed-prometheus-pod-identity-test"
+)
+
+// Workload Identity test vars
+var (
+	testNamespaceWI          = fmt.Sprintf("%s-ns", testNameWorkloadIdentity)
+	deploymentNameWI         = fmt.Sprintf("%s-deployment", testNameWorkloadIdentity)
+	monitoredAppNameWI       = fmt.Sprintf("%s-monitored-app", testNameWorkloadIdentity)
```

Comment:
please, use a random name. I request this because eventually the test will be running in both clusters (PR and main) at the same time and if your query is only based on the app name, both tests could generate conflicts between them. This doesn't happen in other prometheus tests because they are internal, but this prometheus is external and shared between the clusters. 
You could just add a random number at the end of the name with this:
```suggestion
	monitoredAppNameWI       = fmt.Sprintf("%s-monitored-app"-%d, testNameWorkloadIdentity, GetRandomNumber())
```

Reply:
nit: `"%s-monitored-app-%d"`

Reply:
updated!

Reply:
done